sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/WrapperSdkExceptionManager.java

```diff
@@ -0,0 +1,133 @@
+package com.microsoft.azure.mobile.crashes;
+
+import com.microsoft.azure.mobile.crashes.ingestion.models.ManagedErrorLog;
+import com.microsoft.azure.mobile.crashes.utils.ErrorLogHelper;
+import com.microsoft.azure.mobile.utils.MobileCenterLog;
+import com.microsoft.azure.mobile.utils.storage.StorageHelper;
+
+import org.json.JSONException;
+
+import java.io.File;
+import java.util.UUID;
+import java.util.Map;
+import java.util.HashMap;
+import java.io.IOException;
+
+
+public class WrapperSdkExceptionManager {
+    /**
+     * File extension for data files created by this class.
+     */
+    private static final String DATA_FILE_EXTENSION = ".dat";
+
+    /**
+     * Contains wrapper SDK data that has been loaded into memory
+     */
+    private static Map<String, byte[]> sWrapperExceptionDataContainer = new HashMap<>();
+
+    /**
+     * Store the in-memory wrapper exception data to disk. This should only be used by a wrapper SDK.
+     *
+     * @param data  The data to be saved
+     * @param reportIdString  The associated error UUID string
+     */
+    public static void saveWrapperExceptionData(byte[] data, String reportIdString) {
+        if (reportIdString == null) {
+            MobileCenterLog.error(Crashes.LOG_TAG, "Trying to save wrapper exception data with null UUID");
+            return;
+        }
+
+        sWrapperExceptionDataContainer.put(reportIdString.toString(), data);
+        File dataFile = getFile(reportIdString.toString());
+        try {
+            StorageHelper.InternalStorage.writeObject(dataFile, data);
+        }
+        catch (IOException e) {
+            MobileCenterLog.error(Crashes.LOG_TAG, "Error writing wrapper exception data to file", e);
+        }
+    }
+
+    /**
+     * Delete wrapper exception data from disk and store it in memory
+     *
+     * @param reportId    The associated error UUID
+     */
+    public static void deleteWrapperExceptionData(UUID reportId) {
+        if (reportId == null) {
+            MobileCenterLog.error(Crashes.LOG_TAG, "Trying to delete wrapper exception data with null UUID");
+            return;
+        }
+
+        File dataFile = getFile(reportId.toString());
+        byte[] loadResult = loadWrapperExceptionData(reportId.toString());
+        if (loadResult == null) {
+            MobileCenterLog.error(Crashes.LOG_TAG, "Trying to delete wrapper exception data with null UUID");
```

Comment:
Is this a write log string? null UUID is not possible, you are checking null at the first line.
And, report ID instead of UUID?